flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  description = "A flake for s2n-tls";
+
+  inputs.nixpkgs.url = github:NixOS/nixpkgs/nixos-22.11;
+
+  outputs = { self, nix, nixpkgs, flake-utils }: 
+     flake-utils.lib.eachDefaultSystem (system:
+      let pkgs = nixpkgs.legacyPackages.${system};
+   in rec {
+      packages.s2n-tls = pkgs.stdenv.mkDerivation {
+        src = self;
+        name = "s2n-tls";
+        inherit system; 
+        # TODO: override the s2n-tls version
+
+        buildInputs = [ pkgs.cmake
+                        pkgs.openssl_3 ]; # s2n-config has find_dependency LibCrypto
+
+        cmakeFlags = [
+            "-DBUILD_SHARED_LIBS=ON"
+            "-DUNSAFE_TREAT_WARNINGS_AS_ERRORS=OFF" # disable -Werror
```

Comment:
We should be checking for these errors in our CI so we should have a way to turn it on.

Reply:
Currently passing on gcc11, so I'll remove this.